lib/rubocop/cop/layout/class_structure.rb

```diff
@@ -146,6 +146,10 @@ class ClassStructure < Cop
           (send nil? { :private :protected :public })
         PATTERN
 
+        def_node_matcher :dynamic_constant?, <<~PATTERN
```

Comment:
Probably it'd be better to just skip such nodes in the auto-correct instead of avoiding the offense for them.